### PR TITLE
Schema: Adjust primary key irregularities of icon_image, notes_url & …

### LINKED ISSUE #450: Primary key of icon_image, action_url, notes_url should not explicitly include the environment_id column
These three tables have `PRIMARY KEY (environment_id, id)` and the comment for `id` says that the environment is not included in the hash:
https://github.com/Icinga/icingadb/blob/dece98e3e9e12bd592e2ea5d59ac6cb6f6a3ed23/schema/mysql/schema.sql#L722-L747

However, the code that generates the IDs actually includes the environment ID:
https://github.com/Icinga/icinga2/blob/fb2a58f177eddde909261e3da06894e28c889d3f/lib/icingadb/icingadb-objects.cpp#L682-L729

So we can just remove this irregularity from the schema without other code changes.

### PR DESCRIPTION
**Pull request by yhabteab**
…action_url tables

fixes #450

### CODE REVIEW

**Review by julianbrost — CHANGES_REQUESTED** [review]
All the other comments use `environment.id` instead of `environment_id`, so this should stay consistent.

Also, for MySQL, this has to be added to the upgrade file (but not for PostgreSQL as there's no released version of the schema yet).

**Comment by yhabteab** [comment]
Should the pk always have to be created with a constraint? Like `add constraint pk_xy pk ..`

File: schema/mysql/upgrades/1.0.0.sql [hunk 1]
```diff
@@ -26,6 +26,23 @@ ALTER TABLE flapping_history
...
+  DROP COLUMN id,
+  DROP PRIMARY KEY,
+  ADD COLUMN id binary(20) NOT NULL COMMENT 'sha1(environment.id + action_url)' FIRST,
+  ADD PRIMARY KEY (id);
+
+ALTER TABLE notes_url
+  DROP COLUMN id,
+  DROP PRIMARY KEY,
+  ADD COLUMN id binary(20) NOT NULL COMMENT 'sha1(environment.id + notes_url)' FIRST,
+  ADD PRIMARY KEY (id);
```
Comment (julianbrost) [inline schema/mysql/upgrades/1.0.0.sql:45]: Will this work if there is data in the tables?

**Comment by julianbrost** [comment]
> Should the pk always have to be created with a constraint? Like `add constraint pk_xy pk ..`

I'm not sure what this question is about. Given that you mention explicitly named constraints, I think you're referring to PostgreSQL but that change seems to be fine. In case your question is if we always do these named explicitly for PostgreSQL, then yes, we need to refer to these constraints in https://github.com/Icinga/icingadb/blob/89bcc99071fa23667efc45aaf2b981fa3c3d5cac/pkg/icingadb/db.go#L198-L200

File: schema/mysql/upgrades/1.0.0.sql (hunk 1 above)
Comment (yhabteab) [reply schema/mysql/upgrades/1.0.0.sql]: Nope, they will lose their hash IDs. Fixed!

**Comment by yhabteab** [comment]
> I'm not sure what this question is about.

My question is referring to whether I should explicitly set the pk name in the upgrade script or not!

**Comment by julianbrost** [comment]
For MySQL? We don't name them in the full schema, so no need to do so in the upgrade. (Is that even something MySQL would support?)

The current version of the PR seems fine (i.e. I don't have any more complaints right now, I just have to test it).

**Review by julianbrost — APPROVED** [review]

**Comment by julianbrost** [comment]
@yhabteab Can this be merged immediately or should we rather wait for https://github.com/Icinga/icingadb-web/pull/525 to also be ready?

**Comment by yhabteab** [comment]
It's already approved, so if you want to merge this, feel free.

**Pull request merged**